### ISSUE TITLE
Missing escape of backslashes in prompt.sub

### ISSUE THREAD

**Issue by assaftibm** [IBM/ibm-generative-ai#83]
### Version Information
 - Package Version: 0.1.19
 - Operating System: MacOS 13.3.1

### What is the expected behavior?
Support for backslashes in prompt field values

### What is the actual behavior?
Backslashes are unsupported

### Please provide a unit test that demonstrates the bug.
```
  prompt = PromptPattern.from_str("{{context}}\nquestion: {{question}}\nanswer:")
  prompt.sub("context", "Copy the file to \Conf")
```

### Other notes on how to reproduce the issue?
The above code causes re.sub() to raise an exception as it attempts to interpret the \C as a regex meta-character.
The user isn't aware of prompt.sub() using re.sub() so he/she is not aware of the need to escape backslashes.

### Any possible solutions?
In prompt_template.py, line [204](https://github.com/IBM/ibm-generative-ai/blob/a7a85af4249dcd346ea816dd3126e60121155a2d/src/genai/prompt_pattern.py#L204), add escaping of backslashes as described at the bottom of the documentation of  re.escape() in the [Python docs](https://docs.python.org/3/library/re.html#:~:text=This%20function%20must%20not%20be%20used%20for%20the%20replacement%20string%20in%20sub()%20and%20subn()%2C%20only%20backslashes%20should%20be%20escaped.%20For%20example%3A) .

This means that line 204:
```
        self.dump = re.sub(pattern, value, prompt)
```
Should be changed into:
```
        self.dump = re.sub(pattern, value.replace('\\', r'\\'), prompt)
```

### Can you identify the location in the GENAI source code where the problem exists?
In prompt_template.py, line [204](https://github.com/IBM/ibm-generative-ai/blob/a7a85af4249dcd346ea816dd3126e60121155a2d/src/genai/prompt_pattern.py#L204)

### If the bug is confirmed, would you be willing to submit a PR?

Yes

**Comment by onkarbhardwaj** [comment]
@assaftibm : Thanks for your willingness to submit a PR! Assigning this issue to you. Our guidelines are here: [link](https://github.com/IBM/ibm-generative-ai/blob/main/DEVELOPMENT.md)

**Comment by assaftibm** [comment]
@onkarbhardwaj I created the PR. Thanks.

**Comment by Tomas2D** [comment]
Released in [v0.2.6](https://github.com/IBM/ibm-generative-ai/releases/tag/v0.2.6) 🚀 

Thanks! 🥇
